src/archive.h

```diff
@@ -39,19 +36,86 @@ class QString;
 
 class FileData {
 public:
-  virtual QString getFileName() const = 0;
-  virtual void addOutputFileName(QString const &fileName) = 0;
-  virtual std::vector<QString> getAndClearOutputFileNames() = 0;
+
+  /**
+   * @return the full filepath to this entry.
+   */
+  virtual std::wstring getFileName() const = 0;
+
+  /**
+   * @return the size of this entry in bytes (uncompressed).
+   */
+  virtual uint64_t getSize() const = 0;
+
+  /**
+   * @brief Add the given filename to the list of files to create from this
+   *   entry when extracting.
+   *
+   * @param filepath The filename to add.
+   */
+  virtual void addOutputFileName(std::wstring const& filepath) = 0;
+  
+  /**
+   * @brief Retrieve the list of filepaths to extract this entry to and clear the
+   *   internal list.
+   *
+   * @return the list of paths this entry should be extracted to.
+   */
+  virtual std::vector<std::wstring> getAndClearOutputFileNames() = 0;

```

Comment:
This seems very impractical for anyone else wanting to use this library. 
Adding a just getOutputFileNames() to just retrieve what is already there would probably beconvenient.
If we want to go the extra mile we could allow to pass a vector of filepahts directly to addOutPutFileName().

Reply:
I kind of agree, I just kept the original functionality. I don't really mind changing it.

I'm not sure having `addOutputFileName` overloads is really useful - It's a two line call with a `for ()` loop, and I'm not sure you would often add multiple output names:

```cpp
for (auto &x: v) file->addOutputFileName(x);
```

And to be "clean", that would probably be a template on `It`, so the call would not be that short: 

```cpp
file->addOutputFileNames(std::begin(v), std::end(v));
```

BTW, if we're changing this, I'm probably going to rename `FileName` to `Path` or `FilePath`.

Reply:
I can also directly return the `vector<std::wstring>&`, and people can do whatever they want. I'm not really against this honestly...

Reply:
Agreed for the vector overload not being that useful. 
For the fileName I was about to go check the implementation to see what it actually is, the comment says full path but my guess was just the path relative to the archive root?

Reply:
Well, `getFileName` has nothing to do with `addOutputFileName` or `getAndClearOutputFileNames` actually since `getFileName` is the path in the archive. But yes, all path are relative: relative to the root of the archive or to the extraction folder.

Well, actually I don't really now how `getFileName` behave when the archive contains absolute path but that's the same behavior as before.

I was pretty sure to have documented this somewhere but I'll change the function documentations.

Reply:
It's specified in the README actually, but better put it here also.

Reply:
> I can also directly return the `vector<std::wstring>&`, and people can do whatever they want. I'm not really against this honestly since there is no reason not to allow users to modify the vector of filepaths.

Nah I prefer encapsulation the operations on it in functions in case we ever need to make this threadsafe or change the behavior in some other way.

Reply:
I've updated `FileData` to use `path` instead of `name`, be clear in the comments that those are relative path (to the output folder), split `getAndClear` in `get` and `clear`.

Reply:
Nice work!